pages/dashboard/DashboardHandlerNext.php

```diff
@@ -0,0 +1,358 @@
+<?php
+/**
+ * @file pages/dashboard/DashboardHandlerNext.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DashboardHandlerNext
+ *
+ * @ingroup pages_dashboard
+ *
+ * @brief Handle requests for user's dashboard.
+ */
+
+namespace PKP\pages\dashboard;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\handler\Handler;
+use APP\template\TemplateManager;
+use PKP\components\forms\dashboard\SubmissionFilters;
+use PKP\core\JSONMessage;
+use PKP\core\PKPApplication;
+use PKP\core\PKPRequest;
+use PKP\db\DAORegistry;
+use PKP\plugins\Hook;
+use PKP\security\authorization\PKPSiteAccessPolicy;
+use PKP\security\Role;
+use PKP\submission\Collector as SubmissionCollector;
+use PKP\submission\GenreDAO;
+use PKP\submission\PKPSubmission;
+
+define('SUBMISSIONS_LIST_ACTIVE', 'active');
+define('SUBMISSIONS_LIST_ARCHIVE', 'archive');
+define('SUBMISSIONS_LIST_MY_QUEUE', 'myQueue');
+define('SUBMISSIONS_LIST_UNASSIGNED', 'unassigned');
+
+class DashboardHandlerNext extends Handler
+{
+    /** @copydoc PKPHandler::_isBackendPage */
+    public $_isBackendPage = true;
+
+    public int $perPage = 30;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->addRoleAssignment(
+            [Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_AUTHOR, Role::ROLE_ID_REVIEWER, Role::ROLE_ID_ASSISTANT],
+            ['index', 'tasks', 'myQueue', 'unassigned', 'active', 'archives']
+        );
+    }
+
+    /**
+     * @copydoc PKPHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        $this->addPolicy(new PKPSiteAccessPolicy($request, null, $roleAssignments));
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * Display about index page.
+     *
+     * @param PKPRequest $request
+     * @param array $args
+     */
+    public function index($args, $request)
+    {
+        $context = $request->getContext();
+        $dispatcher = $request->getDispatcher();
+
+        if (!$context) {
+            $request->redirect(null, 'user');
+        }
+
+        $templateMgr = TemplateManager::getManager($request);
+        $this->setupTemplate($request);
+
+        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_USER_ROLES);
+        $apiUrl = $dispatcher->url($request, PKPApplication::ROUTE_API, $context->getPath(), '_submissions');
+
+        $sections = Repo::section()
+            ->getCollector()
+            ->filterByContextIds([$context->getId()])
+            ->getMany();
+
+        $categories = Repo::category()
```

Comment:
There's definitely going to be optimization work required on this page; we're loading all categories, sections, user groups, and potentially all queued submissions into memory for each page load.

Categories, for example, could be quite a long list -- but I don't think they'll frequently be relevant to the editorial workflow.